ydb/aio/query/session.py

```diff
@@ -0,0 +1,138 @@
+import asyncio
+
+from typing import (
+    Optional,
+)
+
+from .base import AsyncResponseContextIterator
+from .transaction import QueryTxContextAsync
+from .. import _utilities
+from ... import issues
+from ..._grpc.grpcwrapper import common_utils
+from ..._grpc.grpcwrapper import ydb_query_public_types as _ydb_query_public
+
+from ...query import base
+from ...query.session import (
+    BaseQuerySession,
+    QuerySessionStateEnum,
+)
+
+
+class QuerySessionAsync(BaseQuerySession):
+    """Session object for Query Service. It is not recommended to control
+    session's lifecycle manually - use a QuerySessionPool is always a better choise.
+    """
+
+    _loop: asyncio.AbstractEventLoop
+    _status_stream: _utilities.AsyncResponseIterator = None
+
+    def __init__(
+        self,
+        driver: common_utils.SupportedDriverType,
+        settings: Optional[base.QueryClientSettings] = None,
+        loop: asyncio.AbstractEventLoop = None,
+    ):
+        super(QuerySessionAsync, self).__init__(driver, settings)
+        self._loop = loop if loop is not None else asyncio.get_running_loop()
+
+    async def _attach(self) -> None:
+        self._stream = await self._attach_call()
+        self._status_stream = _utilities.AsyncResponseIterator(
+            self._stream,
+            lambda response: common_utils.ServerStatus.from_proto(response),
+        )
+
+        first_response = await self._status_stream.next()
+        if first_response.status != issues.StatusCode.SUCCESS:
+            pass
+
+        self._state.set_attached(True)
+        self._state._change_state(QuerySessionStateEnum.CREATED)
+
+        self._loop.create_task(self._check_session_status_loop(), name="check session status task")
+
+    async def _check_session_status_loop(self) -> None:
+        try:
+            async for status in self._status_stream:
+                if status.status != issues.StatusCode.SUCCESS:
+                    self._state.reset()
+                    self._state._change_state(QuerySessionStateEnum.CLOSED)
+        except Exception:
+            pass
```

Comment:
Do you need to change state of the session on exception?
Do you need same behaviour for sync method?